File: app/src/main/java/com/denchic45/kts/data/model/firestore/SubjectDoc.java
package com.denchic45.kts.data.model.firestore;

import com.denchic45.kts.data.model.DocModel;

import java.util.List;

public class SubjectDoc implements DocModel {

    protected String id;
    private String name;
    private String iconUrl;
    private String colorName;
    private boolean special;
    private List<String> searchKeys;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getIconUrl() {
        return iconUrl;
    }

    public void setIconUrl(String iconUrl) {
        this.iconUrl = iconUrl;
    }

    public String getColorName() {
        return colorName;
    }

    public void setColorName(String colorName) {
        this.colorName = colorName;
    }

    public boolean isSpecial() {
        return special;
    }

    public void setSpecial(boolean special) {
        this.special = special;
    }

    public List<String> getSearchKeys() {
        return searchKeys;
    }

    public void setSearchKeys(List<String> searchKeys) {
        this.searchKeys = searchKeys;
    }
}

File: app/src/main/java/com/denchic45/kts/data/model/firestore/SubjectDoc.kt
package com.denchic45.kts.data.model.firestore

import com.denchic45.kts.domain.DocModel
import com.denchic45.kts.util.SearchKeysGenerator

class SubjectDoc(
    var id: String,
    var name: String,
    var iconUrl: String,
    var colorName: String,
) : DocModel {
    private constructor() : this(
        "",
        "",
        "",
        ""
    )

    val searchKeys: List<String>
        get() = SearchKeysGenerator().generateKeys(name, String::isNotEmpty)


    companion object {
        fun createEmpty(): SubjectDoc = SubjectDoc()
    }
}